File: AndroidWebBox/hybrid-sdk/src/main/java/com/sniper/hybrid/plugin/HybridPlugin.kt
package com.sniper.hybrid.plugin

import android.content.Intent
import org.json.JSONObject

/**
 * Hybrid插件接口
 * 所有插件必须实现此接口
 */
interface HybridPlugin {
    /**
     * 插件名称，必须唯一
     * 例如: "camera", "gallery", "device"
     */
    fun pluginName(): String

    /**
     * 执行插件方法
     * @param action 动作名称，例如 "capture", "pick"
     * @param params 参数JSON对象
     * @param callback 回调接口
     */
    fun exec(action: String, params: JSONObject, callback: PluginCallback)

    /**
     * 插件被附加时调用
     * @param context 插件上下文
     */
    fun onAttach(context: PluginContext) {}

    /**
     * 插件被分离时调用，清理资源
     */
    fun onDetach() {}

    /**
     * 处理Activity结果
     * @param requestCode 请求码
     * @param resultCode 结果码
     * @param data 数据Intent
     * @return 是否已处理
     */
    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
        return false
    }

    /**
     * 处理权限结果
     * @param requestCode 请求码
     * @param permissions 权限列表
     * @param grantResults 授权结果
     * @return 是否已处理
     */
    fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ): Boolean {
        return false
    }

    /**
     * 获取插件版本
     */
    fun version(): String = "1.0.0"
}

File: iOSWebBox_copy/iOSWebBox/HybridSDK/Plugin/HybridPlugin.swift
import Foundation

/// 插件协议
/// 所有插件必须实现此协议
public protocol HybridPlugin: AnyObject {
    /// 插件上下文
    var pluginContext: PluginContext? { get set }

    /// 插件名称
    func pluginName() -> String

    /// 执行插件方法
    /// - Parameters:
    ///   - action: 方法名
    ///   - params: 参数
    ///   - callback: 回调
    func exec(action: String, params: [String: Any], callback: PluginCallback)

    /// 插件附加到PluginManager时调用
    func onAttach(context: PluginContext)

    /// 插件从PluginManager移除时调用
    func onDetach()

    /// 检查方法是否存在 (默认实现)
    func hasAction(_ action: String) -> Bool
}

public extension HybridPlugin {
    func hasAction(_ action: String) -> Bool {
        // 默认实现,子类可以重写
        return true
    }
}
